zanata-war/src/main/java/org/zanata/security/ZanataIdentity.java

```diff
@@ -260,10 +616,66 @@ public String login(AuthenticationType authType) {
     public String getAccountUsername() {
         HAccount authenticatedAccount =
                 ServiceLocator.instance().getInstance(
-                        JpaIdentityStore.AUTHENTICATED_USER, HAccount.class);
+                        ZanataJpaIdentityStore.AUTHENTICATED_USER, HAccount.class);
         if (authenticatedAccount != null) {
             return authenticatedAccount.getUsername();
         }
         return null;
     }
+
+    public Principal getPrincipal() {
+        return principal;
+    }
+
+    public void checkRestriction(String expression) {
+        log.warn("permission check as EL is not supported yet");
+        throw new UnsupportedOperationException("identity.checkRestriction(exp) is not supported");
```

Comment:
On 2015-07-31 09:51, Patrick Huang wrote:
> In zanata-war/src/main/java/org/zanata/security/ZanataIdentity.java
> <https://github.com/zanata/zanata-server/pull/908#discussion_r35934714>:
> 
>>          if (authenticatedAccount != null) {
>>              return authenticatedAccount.getUsername();
>>          }
>>          return null;
>>      }
>> +
>> +    public Principal getPrincipal() {
>> +        return principal;
>> +    }
>> +
>> +    public void checkRestriction(String expression) {
>> +        log.warn("permission check as EL is not supported yet");
>> +        throw new UnsupportedOperationException("identity.checkRestriction(exp) is not supported");
> 
> You are right we don't use @CheckPermission anywhere. There is no
> @Restrict <https://github.com/Restrict> which uses EL in our code. I
> think at one stage we have migrated all those into explicit call to
> identity.checkPermission. CheckPermission.value can be used as
> permission action name in the future. See
> org/zanata/security/CheckPermissionDecisionVoter.java:59 I've removed
> the method ZanataIdentity.checkRestriction(expression). But in
> ZanatatSecurityInterceptor, it will throw UnsupportedOperationException
> if someone tries to use the value field. Is this alright?

I'm not sure if it's worth keeping around as dead code, but if we do
want to keep it, we should document it properly (eg javadoc on
CheckPermission.value), and remove the bits which won't ever make sense,
like this method.

-- 
Sean Flanigan

Principal Software Engineer
Globalisation Tools Engineering
Red Hat